tests/testutils/utils.go

```diff
@@ -0,0 +1,77 @@
+// This file contains test utils that are general purpose
+// and should one day be moved to a central location for use across all
+// projects.
+package testutils
+
+import (
+	"fmt"
+	"os/exec"
+
+	. "github.com/onsi/gomega"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/projectcalico/felix/fv/containers"
+	"github.com/projectcalico/libcalico-go/lib/api"
+	"github.com/projectcalico/libcalico-go/lib/client"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const vK8sApiserver = "v1.7.3"
+
+func RunK8sApiserver(etcdIp string) *containers.Container {
+	return containers.Run("st-apiserver",
+		fmt.Sprintf("gcr.io/google_containers/hyperkube-amd64:%s", vK8sApiserver),
+		"/hyperkube", "apiserver",
+		"--service-cluster-ip-range=10.101.0.0/16",
+		"--authorization-mode=AlwaysAllow",
+		"--insecure-port=8080",
+		"--insecure-bind-address=0.0.0.0",
+		fmt.Sprintf("--etcd-servers=http://%s:2379", etcdIp),
+	)
+}
+
+func RunEtcd() *containers.Container {
+	return containers.Run("etcd-fv",
+		"quay.io/coreos/etcd",
+		"etcd",
+		"--advertise-client-urls", "http://127.0.0.1:2379",
+		"--listen-client-urls", "http://0.0.0.0:2379")
+}
+
+func GetEtcdClient(etcdIP string) *client.Client {
```

Comment:
nit: I'd name this something like `GetCalicoClient` or similar, given it's not actually an etcd client in the strictest sense.

(There's a real `etcd/client` package that makes this confusing - https://godoc.org/github.com/coreos/etcd/client)